eva-uow/src/main/kotlin/com/razz/eva/uow/UnitOfWorkExecutor.kt

```diff
@@ -63,66 +52,49 @@ class UnitOfWorkExecutor(
     ): RESULT where PRINCIPAL : Principal<*>,
           PARAMS : UowParams<PARAMS>,
           UOW : BaseUnitOfWork<PRINCIPAL, PARAMS, RESULT, *> {
-        val startTime = System.nanoTime()
-        val activeSpan = coroutineContext[ActiveSpanElement]?.span
-        if (activeSpan == null) {
-            logger.debug { "No active span found in uow context, check tracing configuration" }
-        }
-        var timer: Timer? = null
-        var uowSpan: Span? = null
+        lateinit var executorSpan: Span
         lateinit var name: String
         try {
             var currentAttempt = 0
             while (true) {
                 val uow = uowFactory()
                 if (currentAttempt == 0) {
                     name = uow.name()
-                    timer = createTimer(name)
-                    uowSpan = tracer.buildSpan(name).asChildOf(activeSpan).withTag(Tracing.Tags.UOW_NAME, name).start()
+                    executorSpan = executorSpan()
                 }
                 val constructedParams = params(InstantiationContext(currentAttempt))
-                val performSpan = buildPerformSpan(name, uowSpan)
-                val changes = withContext(ActiveSpanElement(performSpan) + PrimaryConnectionRequiredFlag) {
-                    try {
+                val changes = withContext(PrimaryConnectionRequiredFlag + executorSpan.asContextElement()) {
+                    uowSpan(name).use {
                         uow.tryPerform(principal, constructedParams)
-                    } catch (e: Exception) {
-                        performSpan.finishWithError(e)
-                        throw e
                     }
                 }
-                performSpan.finish()
-                val persistSpan = buildPersistSpan(name, uowSpan)
                 val persisted = try {
-                    persisting.persist(
-                        uowName = uow.name(),
-                        params = constructedParams,
-                        principal = principal,
-                        changes = changes.toPersist,
-                        clock = uow.clock(),
-                        uowSupportsOutOfOrderPersisting = uow.configuration().supportsOutOfOrderPersisting
-                    )
+                    withContext(executorSpan.asContextElement()) {
```

Comment:
There were two separate spans before: one for `tryPerform` func and another for changes persisting.